src/components/Button.js

```diff
@@ -1,23 +1,35 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 
+/**
+ * @component Button - the component for each button
+ * @property {Array} selector
+ * @property {String} sym
+ * @property {Function} handleBtnClick
+ */
 class Button extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.handleBtnClick = this.handleBtnClick.bind(this);
+  }
+
+  handleBtnClick() {
+    const { handleBtnClick, sym } = this.props;
+    handleBtnClick(sym);
   }
 
   render() {
     const { selector, sym } = this.props;
     return (
-      <button type="button" className={selector.join(' ')}>{sym}</button>
+      <button type="button" className={selector.join(' ')} onClick={this.handleBtnClick}>{sym}</button>
```

Comment:
[OPTIONAL] This feels to me like it is more complicated than it needs to be. What would you think of using an anonymous function on the button like this: `onClick={() => {props.handleBtnClick(props.sym)}`, of course with deconstruction it would be just `onClick={() => {handleBtnClick(sym)}`. Doing it this way we could skip all of the this.handleBtnClick definition, but I don't know if it might hurt performance. 

Although in such a lightweight application I'm sure it wont hurt much if it does.

Let me know what you think. I really liked the way you structured the calculator. Props! :clap:

Reply:
first, thank you for your comments.
second, I've got a problem with the GitHub desktop and try solving it.